rotkehlchen/data_import/importers/bitcoin_tax.py

```diff
@@ -0,0 +1,285 @@
+
+    def _consume_event(
+            self,
+            cursor: DBCursor,
+            csv_row: dict[str, Any],
+            csv_type: CSVType,
+            timestamp_format: str = '%Y-%m-%d %H:%M:%S %z',
+    ) -> None:
+        """
+        Consumes a single event from a Bitcoin_Tax csv file.
+        Supported file types are: trades, income, spending.
+        Income and spending have identical structure.
+
+        May raise:
+        - UnknownAsset
+        - DeserializationError
+        - UnsupportedCSVEntry
+        """
+        if all(value == '' for value in csv_row.values()):
+            return  # skip empty rows
+
+        # use a deterministic event_identifier to avoid duplicate events in case of reimport
+        event_identifier = f'rotki_events_bitcoin_tax_{hash_csv_row(csv_row)}'
+        timestamp = ts_sec_to_ms(deserialize_timestamp_from_date(
+            date=csv_row['Date'],
+            formatstr=timestamp_format,
+            location='Bitcoin_Tax',
+        ))
+
+        try:
+            location = Location.deserialize(csv_row['Account'])
+        except DeserializationError:
+            location = Location.EXTERNAL
+            if csv_row['Account'] in ('Coinbase Pro', 'GDAX'):
+                location = Location.COINBASEPRO
+
+        asset_resolver = LOCATION_TO_ASSET_MAPPING.get(location, asset_from_common_identifier)
+        base_asset = asset_resolver(csv_row['Symbol'])
+        base_asset_amount = deserialize_asset_amount(csv_row['Volume'])
+        fee_amount = Fee(deserialize_asset_amount(csv_row['Fee'])) if csv_row['Fee'] else Fee(ZERO)  # noqa: E501
+        fee_asset = (
+            asset_resolver(csv_row['FeeCurrency'])
+            if csv_row['FeeCurrency'] and fee_amount is not None else None
+        )
+        action = csv_row['Action']
+        memo = f"Imported description from bitcoin tax: {csv_row['Memo']}" if csv_row['Memo'] else ''  # noqa: E501
+
+        base_asset_balance = AssetBalance(base_asset, Balance(base_asset_amount, ZERO))
+        fee_asset_balance = None
+        if fee_amount != ZERO and fee_asset is not None:
+            fee_asset_balance = AssetBalance(fee_asset, Balance(fee_amount, ZERO))
+
+        if csv_type == 'trades':
+            quote_asset = asset_resolver(csv_row['Currency'])
+            quote_asset_amount = deserialize_asset_amount(csv_row['Cost/Proceeds'])
+            quote_asset_balance = AssetBalance(quote_asset, Balance(quote_asset_amount, ZERO))
+            self._consume_trade_event(
+                cursor=cursor,
+                csv_row=csv_row,
+                event_identifier=event_identifier,
+                timestamp=timestamp,
+                location=location,
+                action=action,
+                base_asset_balance=base_asset_balance,
+                quote_asset_balance=quote_asset_balance,
+                fee_asset_balance=fee_asset_balance,
+                memo=memo,
+            )
+        else:
```

Comment:
That's not what I said. Please read my comment here: https://github.com/rotki/rotki/pull/6157#discussion_r1210189320

I specifically asked to just add a comment with an else since it makes the code more readable. Having the implicit else without any newline space or anything is hard to read. We do this a lot in the code.